src/report/generate_trustymail_report.py

```diff
@@ -0,0 +1,1372 @@
+                for subdomain in domain["subdomains"]:
+                    subdomain_score = self.__score_domain(subdomain)
+                    # if the subdomain has its own dmarc record or if the
+                    # base domain doesn't have a valid dmarc record,
+                    # add subdomain to the subdomain_scores list
+                    if subdomain_score and (
+                        subdomain["dmarc_record"] or not domain["valid_dmarc"]
+                    ):
+                        # add subdomain score to domain's list of
+                        # subdomain_scores
+                        score["subdomain_scores"].append(subdomain_score)
+            return score
+
+        else:  # domain['live'] == "False"
+            # Check if any subdomains of non-live domains support
+            # SMTP; if so, we want to include them in our results, per
+            # CYHY-554)
+            if domain.get("subdomains"):  # if this domain has any subdomains
+                for subdomain in domain["subdomains"]:
+                    if subdomain["domain_supports_smtp"]:
+                        score["has_live_smtp_subdomains"] = True
+                        subdomain_score = self.__score_domain(subdomain)
+                        # If the subdomain has it's own dmarc record,
+                        # add subdomain to the subdomain_scores list.
+                        # No need to check if base domain doesn't have
+                        # a valid DMARC record because base domain is
+                        # not live.
+                        if subdomain_score and subdomain["dmarc_record"]:
+                            # add subdomain score to domain's list of
+                            # subdomain_scores
+                            score["subdomain_scores"].append(subdomain_score)
+            if score["has_live_smtp_subdomains"]:
+                return score
+            else:
+                # only include base domains in the ineligible count;
+                # otherwise lots of non-existent subs will show in the
+                # report
+                if domain["is_base_domain"]:
+                    # NOT CURRENTLY USED?
+                    self.__ineligible_domains.append({"domain": domain["domain"]})
+                return None
+
+    def __populate_report_doc(self):
+        # sort list of all domains
+        self.__all_domains.sort(key=lambda x: x["domain"])
+        # sort list of base domains
+        self.__base_domains.sort(key=lambda x: x["domain"])
+
+        # Go through each base domain and score the attributes
+        for domain in self.__base_domains:
+            score = self.__score_domain(domain)
+            if score:
+                # Add domain's score to master list of scores
+                self.__report_doc["scores"].append(score)
+
+        if not self.__all_eligible_domains_count:
+            # TODO Decide if we want to generate an empty report in
+            # this case
```

Comment:
Reminder to create an issue and mention it in a comment here.

Reply:
Similarly, I don't know why an empty report would be preferred here.  Can I just delete this comment?

Reply:
I'm fine removing this comment- if no one has asked for this by now, I think we can safely continue doing things the same way we have been doing them.

Reply:
See commit 48b9817.

Reply:
Though while we are here, I wouldn't be opposed to changing this `ERROR` message to a `WARNING` message:
https://github.com/cisagov/trustymail_reporter/blob/48b9817d253bb79d4cec979bbacf977e1c1d5057/src/report/generate_trustymail_report.py#L688-L692

Reply:
See commit a9b803a.

Reply:
We are still exiting with a `-1` code here- do you think it's necessary change that to something else?

Reply:
That's how we exit without generating a report.  An alternative would be to make this method return a Boolean indicating success, and then do the same for the `generate_trustymail_report()` method.  That would allow us to move the `sys.exit(-1)` to the `main()` function.  Do you want me to do that?

Reply:
It's your call- I'm ok either way, but I'm leaning towards not making a change like that in a skeletonization PR.

Reply:
I agree wholeheartedly!  I'll create an issue and then we can resolve this jazz.

Reply:
I generalized the issue a bit and created #47.